### PR TITLE
#127 refactor controllers to a standard MVC format

### LINKED ISSUE #127: Refactor controllers to MVC standard
### Describe the feature or problem you'd like to solve

Our architecture is MVC, which should have very slim controllers that should only handle the requests and error handling.

### Proposed solution

Refactor the controllers to use the services and then the services will interact with the models, prepare the data, etc.
A controller should ideally only make 1 call to a service, although in some cases 2 calls makes sense.

### Additional context

N/A

### PR DESCRIPTION
**Pull request by jason-gerard**
## Purpose:
The architecture pattern we are following is MVC, which has very slim controllers, who should only be responsible for handling the routing and error handling

## Changes:
- Refactored the controllers and services to standard MVC
- Added many more typings

close #127

### CODE REVIEW

File: app/forms/signin_form.py
Comment (domsec) [inline app/forms/signin_form.py:17]: This might be something we no longer need as mentioned in https://github.com/LujainKhalaf/Soengram/pull/122#pullrequestreview-591284017

File: app/forms/signin_form.py
Comment (domsec) [inline app/forms/signin_form.py:29]: Similar to above comment

**Review by domsec — COMMENTED** [review]
Everything looks good. We should just confirm with the rest of the group if we in fact need the validation I commented on.

Given how large of a refactor this is and since probably does not include the recent merges, we should probably hold this off until the beginning of Sprint 3. This should be the first thing merged at the start of Sprint 3 and subsequently we should all test the entire site before doing any Sprint 3 development to make sure this refactor didn't introduce any new bugs.

File: app/forms/signin_form.py
Comment (jason-gerard) [reply app/forms/signin_form.py:29]: Fair so in the singin form we won't have any error message to do with length because they should already have a valid email / password of the appropriate length

**Comment by jason-gerard** [comment]
> Everything looks good. We should just confirm with the rest of the group if we in fact need the validation I commented on.
> 
> 
> 
> Given how large of a refactor this is and since probably does not include the recent merges, we should probably hold this off until the beginning of Sprint 3. This should be the first thing merged at the start of Sprint 3 and subsequently we should all test the entire site before doing any Sprint 3 development to make sure this refactor didn't introduce any new bugs.

Yea so we can get it in a ready to merge state then after the demo on friday we can merge it in that way in the case there was a bug it won't effect our demo. sounds good

**Review by domsec — CHANGES_REQUESTED** [review]
Currently getting the following error after either the Sign In or Sign Up. 

image

**Comment by jason-gerard** [comment]
> Currently getting the following error after either the Sign In or Sign Up.
> 
> image

Fixed it, python was not type casting it properly and so in some instances the value was a dict and some it was the actual class. Now I just always store it in the session as a dict, and always retrieve it and re-build it from a dict -> the class.

**Review by domsec — APPROVED** [review]
All features tested and functional.

**Review by lealakkis — APPROVED** [review]

**Pull request merged**
